pom.xml

```diff
@@ -150,6 +150,18 @@
       <version>1.10.19</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-api-mockito</artifactId>
+      <version>1.7.4</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.powermock</groupId>
```

Comment:
Using power mock usually speaks for a design smell. We have been trying to avoid it I believe.

Reply:
hmm, what would be a good way to test the dynamically loaded classes in that case?

Reply:
Implement the interface (probably just a lambda) in test package. In the hook set a flag?

Reply:
The problem is actually loading the implementation class from the test package, so that the ServiceLoader recognizes it. I can't find a straightforward way to do it. It's indeed an indicator that this may not be the best approach.

Reply:
nvm, found what I was missing, albeit might not be the best still